### ISSUE TITLE
templates for calendar items?

### ISSUE THREAD

**Issue by imposeren** [wreckage/django-happenings#10]
Is it possible to use django templates for calendar items? Currently it's very hard to customize how calendar looks.

It may be easy to implement one or other way of customization (I'm willing to make a pull request) but maybe you have some ideas on prefered way? @wreckage, what do you think?

**Comment by wreckage** [comment]
Making it easier to customize the way the calendar looks sounds like a great idea, but I don't have any preferred way of doing it. If you've come up with a way that you believe works, I'd gladly merge in your pull request. The only worry I have is that it may cause some confusion to anybody currently using the project and accustomed to the way customization works now. But an update to the README ought to be enough to inform of the new way.

**Comment by wreckage** [comment]
I've merged in your pull request; the changes you've made look fantastic. Thank you very much for your contributions.

**Comment by wreckage** [comment]
Running the app within a dummy project and using Django Debug Toolbar I'm seeing a lot of db queries being made after merging in the new pull request: before I had 9 queries, afterwards the number swelled to 54. The increase in queries is because each event (as it appears distinctly on the calendar) is requiring a separate query: more events on the calendar means more queries. Is this increase related to the fix in issue #9 ? I really like the improvements you've made, but is there any way to reduce the number of db queries?

**Comment by imposeren** [comment]
i will investigate this later today: this should be not hard to investigate and optimize

**Comment by imposeren** [comment]
On my test project only queries for check_if_cancelled were performed. pull #14 should remove all such queries by using ``_prefetched_objects_cache``. This is undocumented but it's used for long time in django (for example: https://github.com/django/django/commit/662eea116f5a188b6983f5f7c01c8247e3b6b309). Can you give me screenshots or text logs of any other queries that you get on your dummy project (if any remains after #14)

**Comment by wreckage** [comment]
Your fix worked perfectly; my queries shrunk back down. Thank you!
